services/sources/publisher_backlist_service.py

```diff
@@ -158,41 +159,76 @@ def get_records(
-                s3_path = f'{self.title_prefix}/{record_metadata[SOURCE_FIELD][0]}/{file_name}'
-                s3_response = self.s3_manager.putObjectInBucket(file.getvalue(), s3_path, bucket)
-                
-                if not s3_response.get('ResponseMetadata').get('HTTPStatusCode') == 200:
-                    logger.error(f'Failed to upload file for {record_metadata.get("DRB_Record ID")} to S3')
-                    continue
-
-                s3_url = f'https://{bucket}.s3.amazonaws.com/{s3_path}'
                 
                 publisher_backlist_record = PublisherBacklistMapping(record_metadata)
                 publisher_backlist_record.applyMapping()
                 
-                webpub_flags=FileFlags(reader=True, limited_access=True)
-                self.add_has_part_mapping(s3_url, publisher_backlist_record.record, record_permissions['is_downloadable'], record_permissions['requires_login'])
-                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=webpub_flags, path='/publisher_backlist')
+                try:
+                    file_url = self.download_file_from_location(record_metadata, record_permissions, publisher_backlist_record.record)
+                except Exception:
+                    logger.exception(f'Failed to download file for {record}')
+                    continue
                 
+                publisher_backlist_record.record.has_part.append(str(Part(
+                    index=1,
+                    url=file_url,
+                    source=publisher_backlist_record.record.source,
+                    file_type='application/pdf',
+                    flags=str(FileFlags(download=record_permissions['is_downloadable'], nypl_login=record_permissions['requires_login']))
+                )))
+                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=FileFlags(reader=True, nypl_login=record_permissions['requires_login']), path='publisher_backlist')
                 mapped_records.append(publisher_backlist_record)
             except Exception:
                 logger.exception(f'Failed to process Publisher Backlist record: {record_metadata}')
         
         return mapped_records
+    
+    def download_file_from_location(self, record_metadata: dict, record_permissions: dict, record: Record) -> str:
+        file_location = record_metadata.get('DRB_File Location')
+        aws_file_bucket = os.environ.get("FILE_BUCKET", "drb-files-local")
+
+        if not file_location:
+            hath_identifier = next((identifier for identifier in record.identifiers if identifier.endswith('hathi')), None)
+
+            if hath_identifier is not None:
+                hathi_id = hath_identifier.split('|')[0]
+
+                s3 = boto3.client("s3")
+
+                try:
+                    s3.head_object(Bucket=os.environ.get('PDF_BUCKET', None), Key=f'tagged_pdfs/{hathi_id}.pdf')
```

Comment:
it seems like we get the `PDF_BUCKET` key here multiple times with a default, even though all these operations will fail if there is no bucket var.  To be more explicit, I'd suggest putting a block like:

```
pdf_bucket = os.environ["PDF_BUCKET"]
```
This will raise early if the var is missing, which I think is what we want (since the s3 calls will fail anyway in that case).

Reply:
I get your idea and this would help deduplicate the code here too.